content/resources/how-to-guides/late-arriving-data.md

```diff
@@ -0,0 +1,178 @@
+
+
+## Flux scripts in detail: water_level_process.flux
+
+The water_level_process.flux script is an invokable script. It’s responsible for two things:
+
+
+1. Computing the mean value over a parameterized start and stop times. This mean value is written to the water_level_mean bucket. 
+2. Calculating the count or total numper of points over parameterized start and stop times. This count is written to the water_level_checksum bucket. 
+
+The water_leve_process.flux script contains the following code: \
+`// Compute the mean for the window`
+
+
+```
+from(bucket: "water_level_raw")
+    |> range(start: params.start, stop: params.stop)
+    |> mean()
+    |> to(bucket: "water_level_mean", timeColumn: "_stop")
+    |> yield(name: "means")
+
+// Compute and store new checksum for this window
+from(bucket: "water_level_raw")
+    |> range(start: params.start, stop: params.stop)
+    |> group(columns: ["_measurement", "_field", "_stop"])
+    |> count()
+    |> to(bucket: "water_level_checksum", timeColumn: "_stop")
+    |> yield(name: "checksums")
+```
+
+
+
+## Flux scripts in detail: water_level_checsum.flux
+
+The water_level_process.flux script is a task. This task is responsible for:
+
+
+1. Counting the number of points in the water_level_raw bucket (new count) for the last hour across 1minute windows.
+2. Invoking the water_level_process.flux invokable script to calculate a new mean and a new count across 1 minute windows. 
+3. Gathering the previous count in the water_level_checksum bucket (old count) for the last hour. 
+4. Comparing an old count against a new count with a join.  
+5. Filtering for where the counts do not match.
+6. Invoking the water_level_process.flux invokable script to recompute the mean and count for every 1 min window where the counts don't match. 
+
+The task runs at 1m intervals specified by the `every` parameter:
+
+
+```
+option task = {name: "water_level_checksum", every: 1m, offset: 10s}
+```
+
+
+This `every` parameter also determines the window period for which the mean and counts will be computed over. 
+
+The invokeScripts() function is a custom function that invokes the water_level_process.flux invokable script. \
+`invokeScript = (start, stop) =>`
+
+
+```
+    requests.post(
+        // The script ID here has been hardcoded here
+        url: "https://eastus-1.azure.cloud2.influxdata.com/api/v2/scripts/095fabd404108000/invoke",
+
+        headers: ["Authorization": "Token ${token}", "Accept": "application/json", "Content-Type": "application/json"],
+        body: json.encode(v: {params: {start: string(v: start), stop: string(v: stop)}}),
+    )
+```
```

Comment:
I'd parameterize the script ID.

~~~suggestion
```
option task = {name: "water_level_checksum", every: 1m, offset: 10s}

invokeScript = (start, stop, scriptID) =>
    requests.post(
        url: "https://cloud2.influxdata.com/api/v2/scripts/${scriptID}/invoke",

        headers: ["Authorization": "Token ${token}", "Accept": "application/json", "Content-Type": "application/json"],
        body: json.encode(v: {params: {start: string(v: start), stop: string(v: stop)}}),
    )
```
~~~

We also need to instruct them where to find the script ID.